### ISSUE TITLE
Loss function for single points

### ISSUE THREAD

**Issue by oxinabox** [CellBH/DLkitty#3]
From discussion with @augustinas1 and @kaandocal today
the problem with using `-loglikeyhood` as the loss function for single points
is that if for a particular set of conditions there is only one observation (or a bunch of observations that report the same thing) there is nothing that prevents it from reducing the std-dev to zero, except perhaps regularization on the weights -- which would decrease its general freedom and make all the distributions look more similar. This might require a lot of regularization which is problematic.

An alternative we came up with to handle this (as we think this is super common in the datset)
if we have a single point, just minimize the mean.
then make sure to have a little weight regularization which would result in for those case the std-dev being equal to what it is for other conditions in the dataset -- where we do have mean+std-dev to train it off (and so use the mean + stddev loss)
